### ISSUE TITLE
Global Filter is ignored when using Query Builder

### ISSUE THREAD

**Issue by daniel-entitle** [mikro-orm/mikro-orm#3009]
**Describe the bug**
It seems the "applyFilters" logic that happens when using EntityManager.find does not appear to exist when using the query builder.
**Stack trace**
```
No stack trace needed
```

**To Reproduce**
1. Make global Filter `orm.em.addFilter("tenant",(args: any) => ({}))`
2. Query the model with query builder (`await orm.em.createQueryBuilder(ModelToFilterWithGlobalFilter).getResultList()`

**Expected behavior**
Same behavior as EntityManager.find 

**Additional context**
`const temp = await orm.em.createQueryBuilder(TModel).getResultList(); // does not fire global filter`
`const temp2 = await orm.em.find(TModel, {}); //does fire global filter` 
**Versions**

| Dependency | Version |
| - | - |
| node | irrelevant |
| typescript | irrelevant |
| mikro-orm | 5.1 |
| your-driver | postgres |

**Comment by B4nan** [comment]
Yes, because filters are only handled by EM, with QB things need to be explicit.

**Comment by daniel-entitle** [comment]
@B4nan Hey! Is there any workaround for auto filters when using filters? (Because not all queries are available with EM.find)

**Comment by B4nan** [comment]
Not tested, but something like this might work:

```ts
const qb = ...;
const filters = em.applyFilters('EntityName', {}, {}, 'read');
qb.andWhere(filters);
```

**Comment by bozorgmehr96** [comment]
@B4nan, Is there any way to get global filters from EM and apply them to QB? Or maybe an option on `createQueryBuilder()` to apply global filters.
IMO, the fact that they are global makes them different from other filters, and it's better to apply them or at least give the option to apply them on QB, too.

**Comment by B4nan** [comment]
Its a sync method, to apply filters you need async context as they can be async. You can use `em.applyFilters()` to do this.

**Comment by bozorgmehr96** [comment]
Yeah, I ended up doing that.
Also, I think I found a bug on filters; relation filters do not apply on the count query of the `findAndCount()` method. (In the  [JOIN ON](https://mikro-orm.io/docs/filters#filters-and-relationships) condition of the query)
I will test it further and create a new issue for it.

**Comment by B4nan** [comment]
I see, looks like the regular filters work, but not the filters on relations.

**Comment by B4nan** [comment]
I will wait for your repro, but after spending a few hours looking into the count and filters problem, I feel like it actually works correctly. Filters on relations are not considered for the where condition (unless you use `populateWhere: infer`), so it has no effect on the root entity count. I just got it implemented and realized that it only complicates the query without any outcome, as you get left joins that cannot affect the root entity count.
